tiled/serialization/xarray.py

```diff
@@ -0,0 +1,88 @@
+import io
+
+from ..media_type_registration import serialization_registry
+from ..utils import modules_available
+from .dataframe import (
+    APACHE_ARROW_FILE_MIME_TYPE,
+    XLSX_MIME_TYPE,
+    serialize_arrow,
+    serialize_csv,
+    serialize_excel,
+    serialize_html,
+    serialize_parquet,
+)
+
+
+class _BytesIOThatIgnoresClose(io.BytesIO):
+    def close(self):
+        # When the netcdf writer tells us to close(), ignore it.
+        pass
+
+
+if modules_available("scipy"):
+
+    def serialize_netcdf(node, metadata):
+        file = _BytesIOThatIgnoresClose()
+        # This engine is reportedly faster.
+        # Also, by avoiding the default engine, we avoid a dependency on 'scipy'.
+        node.as_dataset().to_netcdf(file, engine="scipy")
```

Comment:
did you mean to change the engine here? this looks like it depends on scipy...

Reply:
It turns out that the scipy engine is the only one that can write to file-like objects (i.e. buffers). So here we use the scipy engine, but this code block runs conditionally on scipy being installed.

Fixed out-of-date comment in 6df6150.